### PR TITLE
i#3728 doxygen docs: Fix x86 build.

### PR DESCRIPTION
**Pull request by hgreving2304**
Patches the broken x86 build. See issue for bigger re-factoring project.

Issue: #3728

### CODE REVIEW

File: core/arch/instr.h [hunk 1]
```diff
@@ -1422,8 +1422,8 @@ DR_API
 /**
  * Returns true iff \p pred denotes a truly conditional predicate: on all
  * architectures, this excludes #DR_PRED_NONE. On ARM it also excludes
- * #DR_PRED_AL and #DR_PRED_OP; on AArch64, it also excludes #DR_PRED_AL
- * and #DR_PRED_NV.
+ * DR_PRED_AL and DR_PRED_OP; on AArch64, it also excludes DR_PRED_AL
```
Comment (derekbruening) [inline core/arch/instr.h]: Hmm, I would vote for instead tweaking the defines so Doxygen sees them.  In fact, it looks like dr_pred_type_t is already set up for Doxygen to see both sides.  It even has a comment at the top about doing that deliberately.  And AARCHXX is predefined.  So why doesn't Doxygen see these defines?  How are you building this "x86 build"?

Comment (hgreving2304) [reply core/arch/instr.h]: I just hacked it and removed "AARCHXX AARCH64" from the PREDEFINED in api/docs/API.doxy
I built regularly 
```
rm -rf *; cmake -DDEBUG=ON -DBUILD_DOCS=ON -DBUILD_TESTS=ON -DCMAKE_EXPORT_COMPILE_COMMANDS=1 -GNinja ../src/
ninja
```

What do these pre-defines mean? Does this mean it expands all macro defs? What about #if/#else then? is that what you were talking about in #3728 ?

Comment (derekbruening) [reply core/arch/instr.h]: I would say that removing from the PREDEFINED list is a broken build we do not support.  These are defines that Doxygen sets when running the pre-processor.  Yes.

Comment (hgreving2304) [reply core/arch/instr.h]: Ok (see #3728 ). Now that I understand what's going on, yes that makes sense.

**Comment by hgreving2304** [comment]
I am closing this. I think we do need to go the route to support the #if/#else, see #3728 .

**Pull request closed without merging**